### PR TITLE
feat(xref): add more consistent search routes

### PR DESCRIPTION
**Pull request by sidvishnoi**
- Make URLs consistent with rest of the backend.
- Make response format inspectable by humans:

## Before:

```ts
type Response = {
  query?: Query[],
  result: [string, Partial<DataEntry>[]][],
}
```

## After:

```ts
type Result = {
  id: string;
  query?: Query;
  result: Partial<DataEntry>[]
};
type PostResponse = { results: Result[] };
```

### `GET /xref/search`

```http
GET /xref/search?term=addEventListener()
```

<details>
<summary>Sample response</summary>

```json
{
  "id": "2c3d7b2203bcd59bb328812bd461e016f9084535",
  "query": {
    "term": "addEventListener()",
    "types": [],
    "id": "2c3d7b2203bcd59bb328812bd461e016f9084535"
  },
  "result": [
    {
      "type": "method",
      "spec": "dom",
      "shortname": "dom",
      "status": "current",
      "uri": "#dom-eventtarget-addeventlistener",
      "normative": true,
      "for": [
        "EventTarget"
      ]
    }
  ]
}
```

</details>

### `POST /xref/search`

```http
POST /xref/search
Content-Type: application/json

 {
  "options": { "all": true, "query": true },
  "queries": [
    { "term": "abort()" }
  ]
 }
```

<details>
<summary>Sample response</summary>

```json
{
  "results": [
    {
      "id": "2c3d7b2203bcd59bb328812bd461e016f9084535",
      "query": {
        "term": "addEventListener()",
        "types": [],
        "id": "2c3d7b2203bcd59bb328812bd461e016f9084535"
      },
      "result": [
        {
          "shortname": "dom",
          "spec": "dom",
          "type": "method",
          "for": ["EventTarget"],
          "normative": true,
          "uri": "#dom-eventtarget-addeventlistener"
        }
      ]
    }
  ]
}
```

</details>

### CODE REVIEW

File: routes/xref/search.get.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  specs?: string | string[];
+  for?: string;
+  type?: string | string[];
+  query?: boolean;
+}
+type IRequest = Request<never, any, never, QueryParams>;
+
+export default async function route(req: IRequest, res: Response) {
+  const term = req.query.term;
+  const forContext = req.query.for;
```
Comment (marcoscaceres) [inline routes/xref/search.get.ts:18]:
```suggestion
  const { term, for: forContext } = req.query;
```

**Review by marcoscaceres — APPROVED** [review]
Looks good.. just a tiny nit.

**Comment by waledo0o23** [comment]
F

**Comment by marcoscaceres** [comment]
@sidvishnoi, this seemed useful... want to rebase it?

**Review by marcoscaceres — APPROVED** [review]
Lgtm

**Pull request merged**